itests/util/src/main/java/org/apache/hadoop/hive/ql/dataset/QTestDatasetHandler.java

```diff
@@ -52,8 +51,8 @@
 
   private File datasetDir;
   private static Set<String> srcTables;
-  private static Set<String> missingTables = new HashSet<>();
```

Comment:
@zabetak : I really want to have this non-static as you did, but I needed to change it in HIVE-22617 as it caused flakyness in TestMTQueries (you can take a look at comments)...however, I'm not really sure if parallel qtest running will be implemented properly in the near future, so TestMTQueries is not a useful unit test

Reply:
Indeed there is a check-then-act race condition here. I was hoping to fix this without making `missingTables` and `tablesToUnload` static but looking at the code that I committed it seems that I screwed up something while I was rebasing :D I will address this in the following commits (hopefully :crossed_fingers: ). Thanks for catching this!